File: front-end/CoaxysProjectTracker/Controllers/TaskController.cs
﻿using CoaxysProjectTracker.Attributes;
using CoaxysProjectTracker.Services;
using CoaxysProjectTracker.Repositories;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.Web;
using System.Web.Mvc;


namespace CoaxysProjectTracker.Controllers
{
    [CustomAuthorize]
    public class TaskController : Controller
    {
        protected TaskRepository repository;

        // GET: Task
        public async Task<ActionResult> Index()
        {
            var tasks = await Api.GetAsync<List<Models.Task>>("task.json");
            return View(tasks);
        }

        public ActionResult Create()
        {
            return View();
        }


        public async Task<ActionResult> Edit(int id)
        {
           
            return View();
        }

        public ActionResult Details()
        { return View(); }

        public async Task<ActionResult> ListAllTasks()
        {
            var tasks = await Api.GetAsync<List<Models.Task>>("task.json");
            return View(tasks);
        }
    }
}

File: back-end/projectTracker/app/controllers/TaskController.java
package controllers;

import models.Task;
import play.mvc.Controller;

import java.util.List;

/**
 * Created by formation04 on 03/04/17.
 */
public class TaskController extends Controller {

    public static void tasks() {
        List<Task> tasks = null;
        try {
            tasks = Task.findAll();
        } catch (Exception e) {
            e.printStackTrace();
        }
        renderJSON(tasks);
    }

    public static void task(Long id) {
        Task task = Task.findById(id);
        renderJSON(task);
    }

}
